core/src/main/java/org/apache/iceberg/ManifestWriter.java

```diff
@@ -116,12 +126,12 @@ void add(ManifestEntry entry) {
    * @param existingFile a data file
    * @param fileSnapshotId snapshot ID when the data file was added to the table
    */
-  public void existing(DataFile existingFile, long fileSnapshotId) {
-    addEntry(reused.wrapExisting(fileSnapshotId, existingFile));
+  public void existing(DataFile existingFile, long fileSnapshotId, long sequenceNumber) {
```

Comment:
nit: I think we need to update the Javadoc.

Reply:
Also, I am using this method in the rewrite manifests action. The entries metadata table will have `sequenceNumber` but its value will be 0, correct?

Reply:
Manifest rewrite uses the `existing(ManifestEntry)` method that copies from the original entry. That will copy the existing entry's sequence number, or 0 if it was null.

Reply:
The new action should use the sequence number from the metadata table. That will be 0 for metadata written without sequence numbers, or the correct sequence number for v2 tables.